ovs/extensions/db/etcd/installer.py

```diff
@@ -197,6 +207,9 @@ def use_external(external, slave_ip, cluster_name):
     def _setup_proxy(initial_cluster, slave_client, cluster_name):
         base_name = 'ovs-etcd-proxy'
         target_name = 'ovs-etcd-{0}'.format(cluster_name)
+        if ServiceManager.has_service(target_name, slave_client) and ServiceManager.get_service_status(target_name, slave_client) is True:
```

Comment:
If calling ```shrink_cluster```, I think that this will always be ```True```, so I'm not sure whether it's still possible to shrink an Etcd cluster now.